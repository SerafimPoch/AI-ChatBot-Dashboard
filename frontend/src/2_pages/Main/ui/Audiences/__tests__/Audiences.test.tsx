import React from "react";
import { render } from "@testing-library/react";
import {
  AudiencesContainer,
  AudiencesTitle,
  AudienceChartContainer,
  AudienceChartPercent,
  AudienceChartLabelWrapper,
  AudienceChartLabel,
  AudienceChartLabelText,
} from "../styles";

describe("Audiences page components", () => {
  it("AudiencesContainer renders correctly", () => {
    const { asFragment } = render(<AudiencesContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudiencesTitle renders correctly", () => {
    const { asFragment } = render(<AudiencesTitle />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudienceChartContainer renders correctly", () => {
    const { asFragment } = render(<AudienceChartContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudienceChartPercent renders correctly", () => {
    const { asFragment } = render(<AudienceChartPercent />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudienceChartLabelWrapper renders correctly", () => {
    const { asFragment } = render(<AudienceChartLabelWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudienceChartLabel renders correctly", () => {
    const { asFragment } = render(<AudienceChartLabel color="green" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("AudienceChartLabelText renders correctly", () => {
    const { asFragment } = render(<AudienceChartLabelText />);
    expect(asFragment()).toMatchSnapshot();
  });
});
