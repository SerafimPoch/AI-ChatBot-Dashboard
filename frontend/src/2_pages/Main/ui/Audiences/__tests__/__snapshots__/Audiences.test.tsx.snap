// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audiences page components AudienceChartContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 32px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudienceChartLabel renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 16px;
  height: 16px;
  border-radius: 4px;
  background-color: green;
}

<div
    class="emotion-0"
    color="green"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudienceChartLabelText renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  color: undefined;
  margin-right: 5px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudienceChartLabelWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudienceChartPercent renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-display);
  font-size: 64px;
  font-weight: 700;
  line-height: 64px;
  color: undefined;
  margin-bottom: 12px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudiencesContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-radius: 32px;
  border: 2px solid #f2f2f2;
  grid-column: 2/3;
  grid-row: 2;
  padding: 24px 16px 24px 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-row: 2;
    grid-column: 2/3;
  }
}

@media (max-width: 1240px) {
  .emotion-0 {
    grid-row: 3;
    grid-column: 1/3;
  }
}

@media (max-width: 1000px) {
  .emotion-0 {
    padding: 24px 16px 24px 24px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Audiences page components AudiencesTitle renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 20px;
  font-weight: 600;
  line-height: 24px;
  color: undefined;
  margin-bottom: 16px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;
